hpx/lcos/detail/future_data.hpp

```diff
@@ -400,9 +400,9 @@ namespace detail
         ///               error description if <code>&ec == &throws</code>.
         virtual result_type* get_result(error_code& ec = throws)
         {
-            if (!get_result_void(ec))
-                return nullptr;
+            if (get_result_void(ec) != nullptr)
             return reinterpret_cast<result_type*>(&storage_);
+            return nullptr;
```

Comment:
This is messed up. Doesn't look right at all.

Reply:
Merge conflict... WIll fix - thanks for spotting.